### PR TITLE
Fixed incorrect printing of matrix brackets

### LINKED ISSUE #9313: Minor bug in pprint when using Latex
```
v_msc= sp.MatrixSymbol('vdot_msc',1,3)
V_msc = sp.diag(*v_msc )
sp.pprint(V_msc)
```

Output:

```
⎡v̇_msc_00      0          0    ⎤
⎢                               ⎥
⎢    0      v̇_msc_01      0    ⎥
⎢                               ⎥
⎣    0          0      v̇_msc_02⎦
```

System: Ipython Jupiter 3.0.0-f75fda4
Python 2.7.9 |Anaconda 2.1.0 (64-bit)| (default, Mar  9 2015, 16:20:48) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]

### LINKED ISSUE #11197: Physics vector printing screws up matrix brackets.
On master:

```
Python 3.5.1 |Continuum Analytics, Inc.| (default, Dec  7 2015, 11:16:01) 
Type "copyright", "credits" or "license" for more information.

IPython 4.2.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from sympy.physics.mechanics.models import n_link_pendulum_on_cart

In [2]: kane = n_link_pendulum_on_cart(1)

In [3]: from sympy import Eq

In [4]: from sympy.physics.mechanics import mechanics_printing

In [5]: me
mechanics_printing  memoryview          

In [5]: mechanics_printing()

In [6]: kane = n_link_pendulum_on_cart(1)

In [7]: Eq(kane.mass_matrix_full * kane.q.col_join(kane.u).diff(), kane.forcing_full)
Out[7]: 
⎡               q₀̇                ⎤ = ⎡          u₀           ⎤
⎢                                  ⎥   ⎢                       ⎥
⎢               q₁̇                ⎥   ⎢          u₁           ⎥
⎢                                  ⎥   ⎢                       ⎥
⎢-l₀⋅m₁⋅cos(q₁)⋅u₁̇ + (m₀ + m₁)⋅u₀̇⎥   ⎢          2            ⎥
⎢                                  ⎥   ⎢- l₀⋅m₁⋅u₁ ⋅sin(q₁) + F⎥
⎢    2                             ⎥   ⎢                       ⎥
⎣  l₀ ⋅m₁⋅u₁̇ - l₀⋅m₁⋅cos(q₁)⋅u₀̇  ⎦   ⎣    g⋅l₀⋅m₁⋅sin(q₁)    ⎦
```

### PR DESCRIPTION
**Pull request by czgdp1807**
#### References to other Issues or PRs

Closes #12205 
Fixes #11197
Fixes #9313

#### Other comments
ping @moorepants 

#### Release Notes

* printing
   * combining accents do no longer count in pretty printing leading to that right hand brackets are not messed up

### CODE REVIEW

**Comment by czgdp1807** [comment]
Results in this PR,
```python
isympy                       
IPython console for SymPy 1.5.dev (Python 3.6.8-64-bit) (ground types: python)

These commands were executed:
>>> from __future__ import division
>>> from sympy import *
>>> x, y, z, t = symbols('x y z t')
>>> k, m, n = symbols('k m n', integer=True)
>>> f, g, h = symbols('f g h', cls=Function)
>>> init_printing()

Documentation can be found at https://docs.sympy.org/dev


In [1]: from sympy.physics.mechanics.models import n_link_pendulum_on_cart                                                              

In [2]: from sympy.physics.mechanics import mechanics_printing                                                                          

In [3]: mechanics_printing()                                                                                                            

In [4]: kane = n_link_pendulum_on_cart(1)                                                                                               

In [5]: Eq(kane.mass_matrix_full * kane.q.col_join(kane.u).diff(), kane.forcing_full)                                                   
Out[5]: 
⎡               q₀̇               ⎤   ⎡          u₀           ⎤
⎢                                ⎥   ⎢                       ⎥
⎢               q₁̇               ⎥   ⎢          u₁           ⎥
⎢                                ⎥   ⎢                       ⎥
⎢-l₀⋅m₁⋅cos(q₁)⋅u₁̇ + (m₀ + m₁)⋅u₀̇⎥ = ⎢          2            ⎥
⎢                                ⎥   ⎢- l₀⋅m₁⋅u₁ ⋅sin(q₁) + F⎥
⎢    2                           ⎥   ⎢                       ⎥
⎣  l₀ ⋅m₁⋅u₁̇ - l₀⋅m₁⋅cos(q₁)⋅u₀̇  ⎦   ⎣    g⋅l₀⋅m₁⋅sin(q₁)    ⎦

In [6]: v_msc= MatrixSymbol('vdot_msc',1,3)                                                                                             

In [7]: V_msc = diag(*v_msc)                                                                                                            

In [8]: V_msc                                                                                                                           
Out[8]: 
⎡v̇_msc_00     0         0    ⎤
⎢                            ⎥
⎢   0      v̇_msc_01     0    ⎥
⎢                            ⎥
⎣   0         0      v̇_msc_02⎦
```

**Comment by czgdp1807** [comment]
Results in the master,

```python
isympy             
IPython console for SymPy 1.5.dev (Python 3.6.8-64-bit) (ground types: python)

These commands were executed:
>>> from __future__ import division
>>> from sympy import *
>>> x, y, z, t = symbols('x y z t')
>>> k, m, n = symbols('k m n', integer=True)
>>> f, g, h = symbols('f g h', cls=Function)
>>> init_printing()

Documentation can be found at https://docs.sympy.org/dev


In [1]: from sympy.physics.mechanics.models import n_link_pendulum_on_cart                                                              

In [2]: from sympy.physics.mechanics import mechanics_printing                                                                          

In [3]: mechanics_printing()                                                                                                            

In [4]: kane = n_link_pendulum_on_cart(1)                                                                                               

In [5]: Eq(kane.mass_matrix_full * kane.q.col_join(kane.u).diff(), kane.forcing_full)                                                   
Out[5]: 
⎡               q₀̇                ⎤   ⎡          u₀           ⎤
⎢                                  ⎥   ⎢                       ⎥
⎢               q₁̇                ⎥   ⎢          u₁           ⎥
⎢                                  ⎥   ⎢                       ⎥
⎢-l₀⋅m₁⋅cos(q₁)⋅u₁̇ + (m₀ + m₁)⋅u₀̇⎥ = ⎢          2            ⎥
⎢                                  ⎥   ⎢- l₀⋅m₁⋅u₁ ⋅sin(q₁) + F⎥
⎢    2                             ⎥   ⎢                       ⎥
⎣  l₀ ⋅m₁⋅u₁̇ - l₀⋅m₁⋅cos(q₁)⋅u₀̇  ⎦   ⎣    g⋅l₀⋅m₁⋅sin(q₁)    ⎦

In [6]: v_msc= MatrixSymbol('vdot_msc',1,3)                                                                                             

In [7]: V_msc = diag(*v_msc)                                                                                                            

In [8]: V_msc                                                                                                                           
Out[8]: 
⎡v̇_msc_00      0          0    ⎤
⎢                               ⎥
⎢    0      v̇_msc_01      0    ⎥
⎢                               ⎥
⎣    0          0      v̇_msc_02⎦
```

**Comment by moorepants** [comment]
This is very nice. I haven't looked at the travis failures, but on the face of this I approve the update. Maybe some other tests are failing now though and more work will be needed to address them. Thanks!

**Comment by czgdp1807** [comment]
I think the original author made some unwanted changes causing tests to fail. I have restored them. Let's see if the tests pass now. Let me know what you think @moorepants

**Comment by moorepants** [comment]
This still needs a unit test to ensure it doesn't regress. Can you add a test?

**Comment by moorepants** [comment]
I'd like to see a test for the `is_combining` function and one for the matrix output.

**Comment by czgdp1807** [comment]
@moorepants I have added a test covering both, `is_combining` and matrix output. Please let me know when can we merge this.

**Comment by moorepants** [comment]
I don't see a test for is_combining.

**Comment by czgdp1807** [comment]
The test added covers `is_combining`. I don't know how this module works, so can't add a test specifically for `is_combining`. Can you suggest one?

File: sympy/printing/pretty/pretty_symbology.py [hunk 1]
```diff
@@ -614,6 +614,20 @@ def annotated(letter):
...
+    """
+    if not pretty_use_unicode():
+        return False
+
+    return True if (u'\N{COMBINING GRAVE ACCENT}' <= sym <=
+                    u'\N{COMBINING LATIN SMALL LETTER X}' or
+
+                    u'\N{COMBINING LEFT HARPOON ABOVE}' <= sym <=
+                    u'\N{COMBINING ASTERISK ABOVE}') else False
+
```
Comment (moorepants) [inline sympy/printing/pretty/pretty_symbology.py]: Can't you make a test that passes in a symbol that meets these criteria? I'm not sure what sym is? Is it a unicode object?

Comment (czgdp1807) [reply sympy/printing/pretty/pretty_symbology.py]: I will look into it. Though it may take some time.

Comment (czgdp1807) [reply sympy/printing/pretty/pretty_symbology.py]: I have added the test. Please take a look.

**Comment by moorepants** [comment]
Looks good to me. If tests pass, I approve merging.

**Comment by czgdp1807** [comment]
@moorepants 
```python
_______________________________________________________________________________
_________ sympy/printing/pretty/tests/test_pretty.py:test_is_combining _________
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/sympy-1.5.dev0-py3.6.egg/sympy/utilities/runtests.py", line 1317, in _timeout
    function()
  File "/home/travis/miniconda/envs/test-environment/lib/python3.6/site-packages/sympy-1.5.dev0-py3.6.egg/sympy/printing/pretty/tests/test_pretty.py", line 6930, in test_is_combining
    [False, True, False, False]
AssertionError
= tests finished: 1013 passed, 1 failed, 11 expected to fail, in 24.70 seconds =
DO *NOT* COMMIT!
Traceback (most recent call last):
  File "<stdin>", line 10, in <module>
Exception: Tests failed
```
I don't know why the above test is failing, I have tried to fix it and it passes locally on my system but it's not working on travis. Can you please help me with this?

**Comment by czgdp1807** [comment]
@sympy/mechanics  Please help me out with the above issue. Thanks.

**Comment by moorepants** [comment]
I think this may be a PY2/PY3 issue. Unicode literals are handled differently. Look at the compatibility module, you may need to use it to prep the inputs to your test function.

**Comment by czgdp1807** [comment]
@oscarbenjamin @smichr Please help me with [this issue](https://github.com/sympy/sympy/pull/17717#issuecomment-544223255). I have tried a lot but unable to figure out why the fixes are not working. The tests pass locally on my system but fail on travis.

**Comment by oscarbenjamin** [comment]
Which version of Python are you using? With 2.7 I can reproduce one of the failures:
```console
$ bin/test sympy/printing/pretty/tests/test_pretty.py -k test_is_combining
==================================================================== test process starts =====================================================================
executable:         /Users/enojb/current/sympy/27venv/bin/python  (2.7.10-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
numpy:              1.15.4
random seed:        5336710
hash randomization: on (PYTHONHASHSEED=631395173)

sympy/printing/pretty/tests/test_pretty.py[1] E                                                                                                         [FAIL]

______________________________________________________________________________________________________________________________________________________________
________________________________________________ sympy/printing/pretty/tests/test_pretty.py:test_is_combining ________________________________________________
Traceback (most recent call last):
  File "/Users/enojb/current/sympy/sympy/sympy/printing/pretty/tests/test_pretty.py", line 6908, in test_is_combining
    line = unicode("v̇_m")
UnicodeDecodeError: 'ascii' codec can't decode byte 0xcc in position 1: ordinal not in range(128)

================================================== tests finished: 0 passed, 1 exceptions, in 1.50 seconds ===================================================
DO *NOT* COMMIT!
```

**Comment by moorepants** [comment]
If you use conda it makes it pretty easy to switch between python 2 and 3 using environments.

**Comment by moorepants** [comment]
> I think this may be a PY2/PY3 issue. Unicode literals are handled differently. Look at the compatibility module, you may need to use it to prep the inputs to your test function.

This is my suggestion.

**Comment by oscarbenjamin** [comment]
Are you sure that you are testing with Python 2.7? The `bin/test` script will run with your default python even if you also have 2.7 installed. The output from the script shows you what version is being used - you can see that mine is 2.7.10 above.

Switching between Python versions is straight-forward using virtual environments. I use virtualenv/venv. It sounds like @moorepants uses conda. Either is fine but you need to make sure that the tests are running with 2.7.

**Comment by czgdp1807** [comment]
I tested the code on my system,
Python 3.6.8
```python
================================================== test process starts ==================================================
executable:         /usr/bin/python3  (3.6.8-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
numpy:              1.14.5
random seed:        5522025
hash randomization: on (PYTHONHASHSEED=137573140)

sympy/printing/pretty/tests/test_pretty.py[132] .........................................................................
...........................................................                                                          [OK]

===================================== tests finished: 132 passed, in 12.12 seconds ======================================
```
Python 2.7.15
```python
================================================== test process starts ==================================================
executable:         /usr/bin/python2  (2.7.15-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       gmpy 1.17
numpy:              1.16.4
random seed:        99427927
hash randomization: on (PYTHONHASHSEED=4249283415)

sympy/printing/pretty/tests/test_pretty.py[132] .........................................................................
..........................................................E                                                        [FAIL]

_________________________________________________________________________________________________________________________
_____________________________ sympy/printing/pretty/tests/test_pretty.py:test_is_combining ______________________________
Traceback (most recent call last):
  File "/home/gagandeep/sympy/sympy/printing/pretty/tests/test_pretty.py", line 6908, in test_is_combining
    line = unicode("v̇_m")
UnicodeDecodeError: 'ascii' codec can't decode byte 0xcc in position 1: ordinal not in range(128)

=============================== tests finished: 131 passed, 1 exceptions, in 3.75 seconds ===============================
DO *NOT* COMMIT!
```
I will fix the failure in Python 2.7.15 but I am still unaware why the other one is happening on travis. Is it due to Python 3.6.3?

**Comment by czgdp1807** [comment]
I have tried using `u_decode`, let's see if it works.

**Comment by czgdp1807** [comment]
> I have tried using `u_decode`, let's see if it works.

Didn't work. I will be grateful for any help on this. Thanks.

**Comment by czgdp1807** [comment]
Any help on this? Is this also due to `symengine` tests?

**Comment by czgdp1807** [comment]
Probably, the tests are failing due to, `export PYTHONIOENCODING=ascii:strict`. Will try it out on my system.

**Comment by czgdp1807** [comment]
The tests are running fine on my system, even with `export PYTHONIOENCODING=ascii:strict`. I don't see any reason of these failing now. Please help me out with this.

**Comment by oscarbenjamin** [comment]
Looking to the last commit that failed (tests haven't finished for the most recent one) the errors seem to be to do with tensorflow. That was fixed elsewhere so maybe there are no errors any more...

**Comment by oscarbenjamin** [comment]
I can reproduce the failure with pytest but not with bin/test:
```console
$ pytest sympy/printing/pretty/tests/test_pretty.py -k test_is_combining
==================================================================== test session starts =====================================================================
platform darwin -- Python 3.7.4, pytest-5.1.2, py-1.8.0, pluggy-0.12.0
architecture: 64-bit
cache:        yes
ground types: gmpy 1.17

rootdir: /Users/enojb/current/sympy/sympy, inifile: pytest.ini
plugins: xdist-1.29.0, forked-1.0.2, instafail-0.4.1, doctestplus-0.4.0
collected 132 items / 131 deselected / 1 selected                                                                                                            

sympy/printing/pretty/tests/test_pretty.py F                                                                                                           [100%]

========================================================================== FAILURES ==========================================================================
_____________________________________________________________________ test_is_combining ______________________________________________________________________

    def test_is_combining():
        line = u("v̇_m")
>       assert [is_combining(sym) for sym in line] == \
            [False, True, False, False]
E       assert [False, False, False, False] == [False, True, False, False]
E         At index 1 diff: False != True
E         Use -v to get the full diff

sympy/printing/pretty/tests/test_pretty.py:6929: AssertionError
                                                                       DO *NOT* COMMIT!                                                                       
============================================================= 1 failed, 131 deselected in 2.15s ==============================================================
```

File: sympy/printing/pretty/pretty_symbology.py [hunk 2]
```diff
@@ -615,6 +615,20 @@ def annotated(letter):
     else:
         return ascii_pics[letter]
 
+def is_combining(sym):
+    """Check whether symbol is a unicode modifier.
+
+    See stringPict.width on usage.
+    """
+    if not pretty_use_unicode():
+        return False
```
Comment (oscarbenjamin) [inline sympy/printing/pretty/pretty_symbology.py]: Why is this here? This is the reason that the test failures are not reproducible across environments. I don't see why this check should change the return value of this function

Comment (czgdp1807) [reply sympy/printing/pretty/pretty_symbology.py]: Agreed. This function shouldn't be concerned be with what `pretty_use_unicode`'s output is. I have removed it and it is working correctly now on `pytest`.

**Comment by czgdp1807** [comment]
@moorepants @oscarbenjamin Can we merge this?

**Comment by moorepants** [comment]
@czgdp1807 Thanks for the diligence here!

**Pull request merged**